test_libs/pyspec/eth2spec/utils/ssz/ssz_typing.py

```diff
@@ -281,6 +281,10 @@ class ElementsType(ParamsMeta):
     length: int
 
 
+class BitElementsType(ElementsType):
+    elem_type = boolean
```

Comment:
Shouldn't this have more of an implementation? Even `def __getitem__(self, i): return (self.items[i // 8] >> (i % 8)) % 2` and the equivalent for setitem?

Reply:
Also, in the phase 0 spec, there is one place where we left-shift a bitfield by 1. The simplest way to left-shift a bitfield would probably be a slice; we'd need code to do a slice.

Reply:
I am not using the integer representation internally, they are just lists of bools. Only at the time of serialization/merkleization the compact representation is used (makes the implementation much cleaner and probably more performant at the same time)

Reply:
Aaah I see. Got it.